src/main/java/seedu/masslinkers/model/student/Student.java

```diff
@@ -204,7 +204,7 @@ public boolean isSameStudent(Student otherStudent) {
         }
 
         return otherStudent != null
-                && otherStudent.getName().equals(getName());
+                && otherStudent.getName().equals(getName()) && otherStudent.getTelegram().equals(getTelegram());
```

Comment:
Should we also include their GitHub? Since GitHub usernames, I believe are also unique.